clang_tidy/run_clang_tidy.sh

```diff
@@ -1,7 +1,23 @@
-set -ue
-
+#! /bin/bash
 # Usage: run_clang_tidy <OUTPUT> [ARGS...]
 
+# --- begin runfiles.bash initialization v2 ---
+# Copy-pasted from the Bazel Bash runfiles library v2.
+set -uo pipefail
+f=bazel_tools/tools/bash/runfiles/runfiles.bash
```

Comment:
hi, thanks for the pr. what does this block do exactly? please add a comment, thanks.

Reply:
The block was inspired from the discussion [here](https://stackoverflow.com/questions/53472993/how-do-i-make-a-bazel-sh-binary-target-depend-on-other-binary-targets) on depending on clang-tidy as an external target (provided by other repo) for remote execution. Since we didn't config clang-tidy as an external dependency,  this block can be removed. Thank you for the review.